src/clr-angular/utils/focus/key-focus/key-focus.spec.ts

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ClrKeyFocus } from './key-focus';
+import { Component, DebugElement, ViewChild, ElementRef, OnInit } from '@angular/core';
+import { ClrKeyFocusModule } from './key-focus.module';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+
+@Component({
+  template: `
+    <button class="btn" #enter>Enter focus</button>
+    <div *ngIf="open" [clrKeyFocus]="direction" [clrFocusOnLoad]="focusOnLoad">
+      <button class="btn">Button 1</button>
+      <button class="btn">Button 2</button>
+      <button class="btn">Button 3</button>
+    </div>
+    <button class="btn">Exit focus</button>
+  `,
+})
+class TestComponent implements OnInit {
+  @ViewChild('enter', { static: true, read: ElementRef })
+  enterElement: ElementRef;
+  @ViewChild(ClrKeyFocus, { static: true })
+  keyFocus: ClrKeyFocus;
+
+  open: boolean = false;
+  direction: string = 'vertical';
+  focusOnLoad = true;
+
+  ngOnInit() {
+    this.enterElement.nativeElement.focus();
```

Comment:
What is the reason behind having the test component set focus to the #enter button?